x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/BasicEnrichTests.java

```diff
@@ -208,6 +214,58 @@ public void testMultiplePolicies() {
         }
     }
 
+    public void testAsyncTaskExecute() throws Exception {
+        String policyName = "async-policy";
+        String sourceIndexName = "async-policy-source";
+
+        {
+            IndexRequest indexRequest = new IndexRequest(sourceIndexName);
+            indexRequest.source("key", "key", "value", "val1");
+            client().index(indexRequest).actionGet();
+            client().admin().indices().refresh(new RefreshRequest(sourceIndexName)).actionGet();
+        }
+
+        EnrichPolicy enrichPolicy =
+            new EnrichPolicy(EnrichPolicy.MATCH_TYPE, null, List.of(sourceIndexName), "key", List.of("value"));
+        PutEnrichPolicyAction.Request request = new PutEnrichPolicyAction.Request(policyName, enrichPolicy);
+        client().execute(PutEnrichPolicyAction.INSTANCE, request).actionGet();
+        ExecuteEnrichPolicyAction.Response executeResponse = client()
+            .execute(ExecuteEnrichPolicyAction.INSTANCE, new ExecuteEnrichPolicyAction.Request(policyName).setWaitForCompletion(false))
+            .actionGet();
+
+        assertThat(executeResponse.getStatus(), is(nullValue()));
+        assertThat(executeResponse.getTaskId(), is(not(nullValue())));
+        GetTaskRequest getPolicyTaskRequest = new GetTaskRequest().setTaskId(executeResponse.getTaskId()).setWaitForCompletion(true);
+        GetTaskResponse taskResponse = client().execute(GetTaskAction.INSTANCE, getPolicyTaskRequest).actionGet();
+        assertThat(((ExecuteEnrichPolicyStatus) taskResponse.getTask().getTask().getStatus()).getPhase(),
+            is(ExecuteEnrichPolicyStatus.PolicyPhases.COMPLETE));
```

Comment:
I would place this inside an `assertBusy(...)`, because there is no guarantee that the task is completed (especially on slow build machines).

Reply:
I've added the assertBusy line. In the code the task request has `wait_for_completion` set to true, is it possible that this call can come back before the task is completed even on slow hardware? I suppose at the very least assertBusy insulates the test against timeout failures

Reply:
But in this test `wait_for_completion` is set to false, right? So it is likely that it returns and then the task isn't completed?

If `wait_for_completion` is true (which is the default) then the api doesn't return until the force merge has completed and therefor the associated task has been completed. So in this case it shouldn't be possible for the task not being in a completed state.

Reply:
Ah nevermind my comment... I missed that `wait_for_completion` is also set on `GetTaskRequest`. Then the task api will not return until the task in question has been completed. I didn't know this.

So the assertBusy(...) doesn't make sense and can be removed.